akka-http-core/src/main/scala/akka/http/scaladsl/model/SslSession.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.model
+
+import java.security.Principal
+import java.security.cert.Certificate
+
+import javax.net.ssl.{ SSLPeerUnverifiedException, SSLSession }
+
+/** Value class to allow access to an SSLSession with Scala types */
+class SslSession(val session: SSLSession) extends AnyVal {
```

Comment:
Do we know the compatibility constraints a value class will impose? Not sure we would gain a lot here with a value class, so maybe this could be a trait instead?